dandiapi/api/views/dandiset.py

```diff
@@ -413,19 +408,24 @@ def users(self, request, dandiset__pk):
             send_ownership_change_emails(dandiset, removed_owners, added_owners)
 
         owners = []
+
         for owner_user in dandiset.owners:
             try:
                 owner_account = SocialAccount.objects.get(user=owner_user)
                 owner_dict = {'username': owner_account.extra_data['login']}
                 if 'name' in owner_account.extra_data:
                     owner_dict['name'] = owner_account.extra_data['name']
+                if request.user.is_authenticated and 'email' in owner_account.extra_data:
+                    owner_dict['email'] = owner_account.extra_data['email']
                 owners.append(owner_dict)
             except SocialAccount.DoesNotExist:
                 # Just in case some users aren't using social accounts, have a fallback
-                owners.append(
-                    {
-                        'username': owner_user.username,
-                        'name': f'{owner_user.first_name} {owner_user.last_name}',
-                    }
-                )
+                owner_dict = {
+                    'username': owner_user.username,
+                    'name': f'{owner_user.first_name} {owner_user.last_name}',
+                }
+                if request.user.is_authenticated:  # Check if user is logged in
+                    owner_dict['email'] = owner_user.email
+                owners.append(owner_dict)
+
```

Comment:
adding to the users function throws a linting error that it is now too complex. I could use some guidance on how best to resolve this. @waxlamp

Reply:
One way to solve this is to always include `email` as a key in the response object, but set it to `None` for a unauthenticated user, and to the email address otherwise. That would do the following:
1. Eliminate the extra control flow complexity that leads to a higher [McCabe score](https://docs.astral.sh/ruff/rules/complex-structure/).
2. Make the response more uniform across the user's login state.

To do this you'd eliminate an if block in favor of a ternary conditional (something like `owner_user.email if request.user.is_authenticated else None`), but I don't know if that counts as a "decision point" to the McCabe algorithm.

Another take: the high complexity score is a clue that we were already doing a lot in this function, and have now tipped over into doing too much. Perhaps a cleaner solution is to add another endpoint that just gets the owner emails. This endpoint could just 403 for unauthenticated callers and unconditionally return email addresses otherwise. The cost would be another function in the codebase and another request at runtime. I'm not sure which approach is better.

A lot of uncertainty in my answer here. We can meet to hash this out -- let me know if you'd like to do so.